syntax = "proto3";

package erp2.protos;

service ERPService {
  // Process incoming orders and return shipping information
  rpc ProcessOrder (OrderRequest) returns (OrderResponse);
  
  // Get product stock information
  rpc GetProductStock (ProductRequest) returns (ProductResponse);
}

message OrderRequest {
  string order_id = 1;
  string customer_id = 2;
  string product_id = 3;
  int32 quantity = 4;
}

message OrderResponse {
  string order_id = 1;
  string shipping_date = 2;
  string order_status = 3;
}

message ProductRequest {
  string product_id = 1;
}

message ProductResponse {
  string product_id = 1;
  string product_name = 2;
  int32 stock_level = 3;
  double retail_price = 4;
}